### ISSUE TITLE
Fix S1854 FP: Pre-increment value is used

### ISSUE THREAD

**Issue by manfred-brands** [SonarSource/sonar-dotnet#9502]
### Description

Rule S1854 doesn't seem to recognize that the [prefix increment operator](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/arithmetic-operators#prefix-increment-operator) value is used.

### Repro steps

```csharp
public IEnumerable<Tuple<double, double>> Select(IDataReader dataReader)
{
    int ordinal = -1;

    int xIndex = ++ordinal;
    int yIndex = ++ordinal;

    while (dataReader.Read())
    {
        yield return Tuple.Create(dataReader.GetDouble(xIndex), dataReader.GetDouble(yIndex));
    }
}
```

### Expected behavior

No issue raised.

### Actual behavior

![image](https://github.com/SonarSource/sonar-dotnet/assets/8655218/6384f62b-0b16-4fa5-b161-e812c7a80a97)

### Known workarounds

Suppress rule or replace last assignment with `...Index = ordinal + 1`.
The problem is that if more fields are decoded later it needs changing instead of adding another `...Index = ++ordinal`

### Related information

Nuget package `SonarAnalyzer.CSharp` version `9.27.0.93347`
.NET 8.0 SDK 
Visual Studio 17.10.3
Windows 10

**Comment by martin-strecker-sonarsource** [comment]
Hello @manfred-brands 

As you already mentioned, S1854 is correctly raised here. We have had complaints about this specific case already in the past and have ticket #8330 about improving the message of S1854 in cases like this.
I see your point about readability/extensibility, but it is only sometimes valid. I would recommend marking this specific case as "Accepted" with your valid explanation that keeping the `++ordinal ` is better in this case.

I'm going to close the issue, but if you don't agree, I will re-open it. Let me know if you think enhancing the message is a good compromise.
